scripts/src/models/orders.ts

```diff
@@ -323,11 +323,15 @@ export type ExternalOrderFactory = OrderFactory & {
 
 function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
 	return Object.assign({}, Order, {
-		"new"(data?: DeepPartial<Order>, ...context: Array<DeepPartial<OrderContext>>): Order {
+		"new"(data?: Mutable<DeepPartial<Order>>, ...context: Array<DeepPartial<OrderContext>>): Order {
 			data = Object.assign(
-				{ nonce: Order.DEFAULT_NONCE },
 				data,
 				{ origin });
+
+			if (typeof data.nonce !== "string") {
```

Comment:
why wouldn't it be string?

Reply:
isNothing

Reply:
I would check if the property exists vs checking it's value, eg `data.hasOwnProperty('nounce')` or `'nounce' in data` if you want to live on the edge (I think it'll be ok).